File: Java/11-Vetores/Vetores.java
class Vetores {
    public static void main(String args[]) {
        // Em java podemos armazenar varios valores em uma unica estrutura
        // chamada vetor ou array.

        // Declarando um vetor
        int[] num = new int[4]; // Vetor possui 4 elementos(Começa do indice 0)
        // Preenchendo vetor
        num[0] = 1;
        num[1] = 3;
        num[2] = 5;
        num[3] = 7; // Ultima posição

        // Tambem pode-se declarar desta forma
        int[] par = { 2, 4, 6, 8 };

        // Vetores podem ser de qualquer tipo de dado, mas suas posições devem conter
        // somente dados do tipo
        // especificado na sua declaração
        double[] decimais = { 1.2, 3.2, 2.5, 1.8, 1.6 };

        // Classes
        String[] nomes = { "Larissa", "Maria", "Paula", "Daniela", "Giselle" };

        Integer[] val = { 2, 1, 6 };

        // Para mostrar um vetor só podemos exibir uma posição por vez
        System.out.println("Vetor num posicao 0 = " + num[0]);
        System.out.println("Vetor num posicao 1 = " + num[1]);

        // Para exibir todos valores é mais simples usar estruturas de repetição
        // onde i é o indice do vetor, portanto deve começar com 0
        System.out.println("Vetor num");
        for (int i = 0; i < 4; i++) {
            System.out.print(num[i] + " ");
            // 4 posições, indices 0, 1, 2 e 3
        }

        // Para facilitar temos o método nomedovetor.length que executa o laço até o
        // tamanho total do vetor
        // mesmo que mudemos o vetor acrescentando ou tirando uma posição o laço não
        // precisa ser alterado
        System.out.println("");
        System.out.println("Vetor par");
        for (int i = 0; i < par.length; i++) {
            System.out.print(par[i] + " ");
        }

        System.out.println("");
        System.out.println("Vetor decimais");
        for (int i = 0; i < decimais.length; i++) {
            System.out.print(decimais[i] + " ");
        }

        System.out.println("");
        System.out.println("Vetor nomes");
        for (int i = 0; i < nomes.length; i++) {
            System.out.print(nomes[i] + " ");
        }

        System.out.println("");
        System.out.println("Vetor val");
        for (int i = 0; i < val.length; i++) {
            System.out.print(val[i] + " ");
        }
        // Todos este vetores possuem tamanhos distintos mas length itera perfeitamente
        // pelos seus elementos
    }
}

File: C/14-Vetores/Vetores.c
#include<stdio.h>

int main(void)
{
    //Vetores são variaveis compostas unidimensionais, em uma unica variavel podemos armazenar
    //diversos dados mas estes dados devem ser do mesmo tipo
    //Para declarar um vetor usamos

    int num[4]; //este vetor tem 4 elementos

    //para atribuir valores manualmente(vetores começam com indice 0)
    num[0] = 2;
    num[1] = 4;
    num[2] = 5;
    num[3] = 8; //este é o ultimo elemento

    //Só podemos mostrar um elemento por vez
    printf("%d ",num[0]);
    printf("%d ",num[1]);
    printf("%d ",num[2]);
    printf("%d \n",num[3]);

    printf("Estrutura de controle \n");

    //Para percorrer estes valores de forma mais simples podemos usar uma estrututa de controle
    for(int i = 0; i < 4; i++) { //o indice começa em 0 e vai até 3 neste vetor
        printf("%d ",num[i]);
    }

    printf("\nEncerando...");

    return 0;
}
